### PR TITLE
Fix deletion of cells when markdown in file

### PR DESCRIPTION
**Pull request by rchiodo**

### CODE REVIEW

File: src/interactiveConcatTextDocument.ts [hunk 1]
```diff
@@ -3,285 +3,333 @@
...
+  Position,
+  Range,
+  TextDocument,
+  Uri,
+  workspace,
+  Event,
+  EventEmitter,
+  Location,
+  TextLine,
+} from "vscode";
```
Comment (rchiodo) [inline src/interactiveConcatTextDocument.ts:16]: I didn't change all that much, prettier just stepped in and updated it to look like the jupyter repo

File: src/notebookConcatDocument.ts [hunk 2]
```diff
@@ -1,302 +1,356 @@
...
+  public getTextDocumentAtPosition(
+    position: Position
+  ): TextDocument | undefined {
+    const location = this.concatDocument.locationAt(position);
+    return this.concatDocument
+      .getComposeDocuments()
+      .find((c) => c.uri === location.uri);
+  }
+
+  private get filteredCells(): NotebookCell[] {
```
Comment (rchiodo) [inline src/notebookConcatDocument.ts:200]: This is where the real fix is. The logic wasn't taking into account that some of the cells don't count.

File: src/test/suite/editingTests.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  NotebookConcatTextDocument,
+  NotebookDocument,
+  Uri,
+} from "vscode";
+import { EnhancedNotebookConcatTextDocument } from "../../nativeNotebookConcatTextDocument";
+import { IVSCodeNotebook } from "../../common/types";
+
+suite("Editing Tests", () => {
+  test("Edit a notebook", () => {
+    withTestNotebook(
```
Comment (rchiodo) [inline src/test/suite/editingTests.test.ts:20]: I think we need a lot more tests to verify this is working correctly. Entered an issue for that:
https://github.com/microsoft/vscode-jupyter/issues/7175

File: README.md [hunk 4]
```diff
@@ -14,10 +14,30 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
...
 contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
 
+### Directions for building
+
+You might want to setup an npm link for the python repository
+
+- git clone https://github.com/microsoft/vscode-python.git
+- git clone https://github.com/microsoft/vscode-jupyter-lsp-middleware.git
+- cd vscode-jupyter-lsp-middleware
+- npm link
```
Comment (rebornix) [inline README.md:24]: 👏 this is my first time seeing npm link, this is really cool.

**Pull request merged**